Dockerfile

```diff
@@ -0,0 +1,49 @@
+# Runs on the default latest node LTS image
+# Build stage
+FROM node:lts AS build
+
+## Create the solid community server directory
+RUN mkdir /community-server
+
+## Set current working directory
+WORKDIR /community-server
+
+## Copy the dockerfile's context's community server files
+COPY . .
+
+## Verify if there are known vulnerabilities in the dependencies
+RUN npm audit --production --audit-level=high
+
+## Install and build the Solid community server
+RUN npm ci
+
+
+# Runtime stage
+FROM node:lts
+
+## Add contact informations for questions about the container
+LABEL maintainer="Solid Community Server Docker Image Maintainer <matthieubosquet@gmail.com>"
+
+## Informs Docker that the container listens on the specified network port at runtime
+EXPOSE 3000
+
+## Set command run by the container
+ENTRYPOINT [ "node", "bin/server.js" ]
+
+## By default run in filemode (overriden if passing alternative arguments)
+CMD [ "-c", "config/config-file.json", "-f", "/data" ]
+
+## Container config & data dir for volume sharing
+## Defaults to filestorage with /data directory (passed through CMD below)
+RUN mkdir /community-server /config /data
+
+## Set current directory
+WORKDIR /community-server
+
+## Copy runtime files from build stage
+COPY --from=build /community-server/package.json .
+COPY --from=build /community-server/bin ./bin
+COPY --from=build /community-server/config ./config
```

Comment:
I think the config path is wrong. You create a dir at `/config` but then use `./config`. Since the WORKDIR is `/community-server` that path expands to `/community-server/config`. I assume you meant `/config`?

I also noticed you are only setting up the config and data directories for mounting. What about the templates directory? Is that something that has to be configured at build-time? If not, we may want to make a `/templates` as well to document it's mountable like data and config.

Reply:
I read somewhere that it was problematic to mount non empty directories from a windows host.
Hence the default two empty mount points (instead of a config mounting point with preloaded default configs).

I don't think the templates dir is really used for anything configurable at the moment. Even the `podTemplateFolder` CLI argument points to `templates/pod`. So I would rather not put it in for the sake of simplicity. I imagine this template folder and its use might evolve in the relatively near future. So I'd rather keep creating a dedicated mounting point for when the use case comes up, another nice little PR will make it possible, so I'm not too worried.

Reply:
Oh interesting, I haven't heard that. Maybe call it `preset-configs` or something to make the distinction clear? In general, I like creating non-empty mount points because then it copies those files into the mounted dir, which is useful in Kubernetes and other orchestration platforms because it means the mounted volume starts off container the default config, including the default values. How will the Solid server handle having config spread across two directories? Are you expecting folks to copy the config files they need to the mounted directory from the preloaded folder, or just reference the preloaded files from the mounted folder? It should be fine either way, I'm mostly asking because my next step after this is merged is to use this container to deploy to Kubernetes and I want to make sure I line up with y'alls plans on how to configure things.

Makes sense regarding the template.

Reply:
If you are only going to use the included configurations, then you don't really need to mount a volume, you can just pass in arguments.

If using a custom config, you could indeed copy the config directory over and do as you please.

But you'll still be constrained in the sense that you could only use the base components that ship with the community server. For more complex use cases really taking advantage of the server's modularity, you'll need to install extra packages, run the build step and so on, so that wouldn't be covered by this image.

Which I think is fine, since the goal here is relative simplicity to cover all the most regular use cases and provide an example to get people started.

Then, for more advance uses, I'm sure people will have no issue either basing from this image and extending it or creating their own from scratch! ;)